src/coreclr/jit/compiler.cpp

```diff
@@ -4947,17 +4953,6 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
                 DoPhase(this, PHASE_OPTIMIZE_VALNUM_CSES, &Compiler::optOptimizeCSEs);
             }
 
-            // Assertion prop can do arbitrary statement remorphing, which
-            // can clone code and disrupt our simpleminded SSA accounting.
-            //
-            // So, disable the ssa checks.
-            //
-            if (fgSsaValid)
-            {
-                JITDUMP("Marking SSA as invalid before assertion prop\n");
-                fgSsaValid = false;
-            }
```

Comment:
@AndyAyersMS The SSA checker is dependent on SSA's dominator tree, which is stale after RBO, so I'm also setting `fgSsaValid = false` in `fgInvalidateDfsTree`. It means we no longer check after CSEs, but I suppose that's not a big issue since RBO is really the last user.